src/perturbopy/postproc/utils/spectra_plots.py

```diff
@@ -0,0 +1,128 @@
+"""
+Utils for ultrafast spectroscopy: pump pulse generation and post-processing
+"""
+
+import os
+import time
+import numpy as np
+import matplotlib.pyplot as plt
+import warnings
+from scipy import special
+
+from perturbopy.io_utils.io import open_hdf5, close_hdf5
+
+from matplotlib.animation import FuncAnimation
+
+from .memory import get_size
+from .timing import TimingGroup
+from .constants import energy_conversion_factor
+
+# Plotting parameters
+from .plot_tools import plotparams
+plt.rcParams.update(plotparams)
+
+
+def plot_occ_ampl(e_occs, elec_kpoint_array, elec_energy_array,
+                  h_occs, hole_kpoint_array, hole_energy_array, pump_energy):
+    """
+    Plot occupation amplitude. Currently hardcoded to the section kz = 0.
+    """
+
+    # find where kz == 0
+
+    fig, ax = plt.subplots(1, 1, figsize=(9, 6))
+
+    # Plot electron occupations
+    idx = np.where((elec_kpoint_array[:, 1] == 0) & (elec_kpoint_array[:, 0] == 0))
+    for i in range(np.size(elec_energy_array, axis=1)):
+        ax.scatter(elec_kpoint_array[idx, 2], elec_energy_array[idx, i], s=0.5, c='black', alpha=0.5)
+        ax.scatter(elec_kpoint_array[idx, 2], elec_energy_array[idx, i], s=e_occs[idx, i][0] * 1000 + 1e-10, c='red', alpha=0.5)
+
+    # Plot hole occupations
+    idx = np.where((hole_kpoint_array[:, 1] == 0) & (hole_kpoint_array[:, 0] == 0))
+    for i in range(np.size(hole_energy_array, axis=1)):
+        ax.scatter(hole_kpoint_array[idx, 2], hole_energy_array[idx, i], s=0.5, c='black', alpha=0.5)
+        ax.scatter(hole_kpoint_array[idx, 2], hole_energy_array[idx, i], s=h_occs[idx, i][0] * 1000 + 1e-10, c='red',
+                   alpha=0.5)
+
+    fsize = 16
+    ax.set_title(f'Occupation amplitude for pump energy {pump_energy:.3f} eV')
+    ax.set_xlabel('Electron Momentum $k_x$')
+    ax.set_ylabel('Energy (eV)')
+
+    plt.show()
+
+
+def animate_pump_pulse(time_step,
+                       elec_delta_occs_array, elec_kpoint_array, elec_energy_array,
+                       hole_delta_occs_array, hole_kpoint_array, hole_energy_array,
+                       pump_energy):
+    """
+    Animate the pump pulse excitation for electrons and holes.
+    """
```

Comment:
In this and previous function more explicit docs is required

Reply:
I've done this, but don't think this is needed, see the comment above.
Still done it to accelerate the PR, so we can discuss later for future developments.